astropy/table/serialize.py

```diff
@@ -144,18 +145,21 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
         if not has_info_class(data, MixinInfo):
             col_cls = MaskedColumn if (hasattr(data, 'mask')
                                        and np.any(data.mask)) else Column
-            new_cols.append(col_cls(data, name=new_name, **new_info))
-            obj_attrs[data_attr] = SerializedColumn({'name': new_name})
-        else:
-            # recurse. This will define obj_attrs[new_name].
-            _represent_mixin_as_column(data, new_name, new_cols, obj_attrs)
-            obj_attrs[data_attr] = SerializedColumn(obj_attrs.pop(new_name))
+            data = col_cls(data, name=new_name, **new_info)
+
+        # Recurse. If this is a Mixin, it will define obj_attrs[new_name],
+        # otherwise it will just add to new_cols and all we have to do is
+        # to add the link to the new name.
+        _represent_mixin_as_column(data, new_name, new_cols, obj_attrs)
+        obj_attrs[data_attr] = SerializedColumn(obj_attrs.pop(new_name,
+                                                              {'name': new_name}))
 
     # Strip out from info any attributes defined by the parent,
     # and store whatever remains.
-    for attr in col.info.attrs_from_parent:
-        if attr in info:
-            del info[attr]
+    if not (isinstance(col, Column) and col.dtype.names):
```

Comment:
This is really an ugly hack; should be possible to make this a bit more elegant!

Reply:
No, it is to ensure that `description`, etc., do not get removed, as they are not stored on the lower-level columns. I actually think the location of this statement is incorrect - it should happen only if there is a single column doing the main representation (such as 'data' for a MaskedColumn)